### PR TITLE
Hide benchmarking CLI behind the feature flag

### LINKED ISSUE #5587: Benchmarking CLI is always compiled but the runtime method `Benchmark_dispatch_benchmark` is not
## Problem

There is a bug in the new benchmarking CLI. While it is always built, the necessary runtime method `Benchmark_dispatch_benchmark` is gated by the option `runtime-benchmarks` which is turned off by default and cannot be provided when building Substrate from the repo root due to the Cargo constraint.

When you use the `benchmark` option, you get something like this:
```
$ cargo run --release -- benchmark --pallet im-online --extrinsic heartbeat
    Finished release [optimized] target(s) in 0.50s
     Running `target/release/substrate benchmark --pallet im-online --extrinsic heartbeat`
2020-04-03 08:53:28 new validator set of size 1 has been elected via ElectionCompute::OnChain for era 0
Error: Input("Error executing runtime benchmark: MethodNotFound(\"Benchmark_dispatch_benchmark\")")
```

## Solution

Perhaps forcing `runtime-benchmarks` as standard for the CLI, that is:
```
diff --git a/bin/node/cli/Cargo.toml b/bin/node/cli/Cargo.toml
index 9842fd2ee..982eab209 100644
--- a/bin/node/cli/Cargo.toml
+++ b/bin/node/cli/Cargo.toml
@@ -133,7 +133,7 @@ path = "../../../client/cli"
 optional = true
 
 [features]
-default = [ "cli" ]
+default = [ "cli", "runtime-benchmarks" ]
 browser = [
        "browser-utils",
        "wasm-bindgen",
```

### PR DESCRIPTION
**Pull request by DarkEld3r**
Closes #5587.

### CODE REVIEW

**Comment by bkchr** [comment]
I assume your tried it once? :D

**Comment by DarkEld3r** [comment]
Sure, the `cargo run --release -- benchmark ...` is resulting in the following error:
```
error: Found argument 'benchmark' which wasn't expected, or isn't valid in this context
```
Is it OK? Perhaps we can give a more helpful message.

File: bin/node/cli/src/command.rs [hunk 1]
```diff
@@ -79,6 +79,7 @@ pub fn run() -> Result<()> {
 
 			runner.sync_run(|config| cmd.run::<Block, RuntimeApi, Executor>(config))
 		}
+		#[cfg(feature = "runtime-benchmarks")]
```
Comment (bkchr) [inline bin/node/cli/src/command.rs:82]: Yeah, I think it is a good idea to improve the message. We should remove the cfg here and in the enum declaration.

If the feature is given, it should call the code below and if the feature is not given, it should print that benchmarking wasn't enabled when building the node.

**Comment by bkchr** [comment]
Nice ty!

**Comment by vkomenda** [comment]
I would have been even better if you gated the whole `benchmark` command instead of printing an info message. Like

```
        #[cfg(feature = "runtime-benchmarks")]
        Some(Subcommand::Benchmark(cmd)) => { ...
```

**Comment by bkchr** [comment]
Why? The next day someone would open an issue and ask why the benchmark command is unknown.

**Comment by vkomenda** [comment]
> Why? The next day someone would open an issue and ask why the benchmark command is unknown.

@bkchr that depends on how Parity documents CLI features. A readable explanation in README.md would suffice for people to see there is a runtime benchmarking option.

The next issue might be for the lack of docs ;-)

**Pull request merged**